internal/controller/config/config.go

```diff
@@ -15,17 +15,19 @@ const (
 )
 
 type Config struct {
-	Debug            bool          `mapstructure:"debug"`
-	AgentTokenSecret string        `mapstructure:"agent-token-secret" validate:"required"`
-	BuildkiteToken   string        `mapstructure:"buildkite-token"    validate:"required"`
-	Image            string        `mapstructure:"image"              validate:"required"`
-	JobTTL           time.Duration `mapstructure:"job-ttl"`
-	MaxInFlight      int           `mapstructure:"max-in-flight"      validate:"min=0"`
-	Namespace        string        `mapstructure:"namespace"          validate:"required"`
-	Org              string        `mapstructure:"org"                validate:"required"`
-	Tags             stringSlice   `mapstructure:"tags"               validate:"min=1"`
-	ProfilerAddress  string        `mapstructure:"profiler-address"   validate:"omitempty,hostname_port"`
-	ClusterUUID      string        `mapstructure:"cluster-uuid"       validate:"omitempty"`
+	Debug                bool                   `mapstructure:"debug"`
+	AgentTokenSecret     string                 `mapstructure:"agent-token-secret" validate:"required"`
+	BuildkiteToken       string                 `mapstructure:"buildkite-token"    validate:"required"`
+	Image                string                 `mapstructure:"image"              validate:"required"`
+	JobTTL               time.Duration          `mapstructure:"job-ttl"`
+	MaxInFlight          int                    `mapstructure:"max-in-flight"      validate:"min=0"`
+	Namespace            string                 `mapstructure:"namespace"          validate:"required"`
+	Org                  string                 `mapstructure:"org"                validate:"required"`
+	Tags                 stringSlice            `mapstructure:"tags"               validate:"min=1"`
+	ProfilerAddress      string                 `mapstructure:"profiler-address"   validate:"omitempty,hostname_port"`
+	ClusterUUID          string                 `mapstructure:"cluster-uuid"       validate:"omitempty"`
+	PodSpecPatch         map[string]interface{} `mapstructure:"pod-spec-patch"     validate:"omitempty"`
```

Comment:
What happens if this is specified as something k8s would not recognise like:
```yaml
llamas: true
```

Is it a no-op, or is there a failure of the patch to apply which causes the Job to fail to start? If it's the latter, then I suggest we should try to detect this as earlier as possible.

It's most likely too onerous to validate it in the JSON Schema for the helm chart, but I think if some Kubernetes library exports as type for this, we should try to marshal into that so that the helm chart fails to deploy if `podSpecPatch` would cause and error.

Reply:
Oh yeah, both this and #248 will need to update to the Helm chart's [JSONSchema](https://github.com/buildkite/agent-stack-k8s/blob/-/charts/agent-stack-k8s/values.schema.json#L208)